run_unittests.py

```diff
@@ -3232,15 +3235,52 @@ def dist_impl(self, vcs_init, vcs_add_all=None, include_subprojects=True):
             if vcs_add_all:
                 vcs_add_all(self.create_dummy_subproject(project_dir, 'samerepo'))
             self.init(project_dir)
-            self.build('dist')
-            self.assertPathExists(xz_distfile)
-            self.assertPathExists(xz_checksumfile)
-            self.assertPathDoesNotExist(zip_distfile)
-            self.assertPathDoesNotExist(zip_checksumfile)
-            self._run(self.meson_command + ['dist', '--formats', 'zip'],
-                      workdir=self.builddir)
-            self.assertPathExists(zip_distfile)
-            self.assertPathExists(zip_checksumfile)
+            self.build()
+
+            for i in range(len(archive_choices) + 1):
+                permutations = tuple(itertools.permutations(archive_choices, i))
+                for i, subset in enumerate(permutations):
+                    # skip empty subset
+                    if len(subset) == 0:
```

Comment:
using `len()` to test for container emptiness is a python anti-pattern, just:
```python
if not subset:
```

Reply:
Good to know.